docs/uri_backends.rst

```diff
@@ -1,31 +1,63 @@
 URI Schemes and Backends
 ========================
 
+BaseURIBackend
+--------------
+
+``Py-EthPM`` uses the ``BaseURIBackend`` as the parent class for all of its URI backends. To write your own backend, it must implement the following methods. 
+
+.. py:method:: BaseURIBackend.can_resolve_uri(uri)
+
+   Returns a bool indicating whether or not this backend is capable of resolving the given URI to a manifest.
+   A content-addressed URI pointing to valid manifest is said to be capable of "resolving".
+
+.. py:method:: BaseURIBackend.can_translate_uri(uri)
+
+   Returns a bool indicating whether this backend class can translate the given URI to a corresponding content-addressed URI.
+   A registry URI is said to be capable of "transalating" if it points to another content-addressed URI in its respective on-chain registry.
+
+.. py:method:: BaseURIBackend.fetch_uri_contents(uri)
+
+   Fetches the contents stored at the provided uri, if an available backend is capable of resolving the URI. Validates that contents stored at uri match the content hash suffixing the uri.
+
+
 IPFS
 ----
 
 ``Py-EthPM`` has multiple backends available to fetch/pin files to IPFS. The desired backend can be set via the environment variable: ``ETHPM_IPFS_BACKEND_CLASS``.
 
 - ``InfuraIPFSBackend`` (default)
     - `https://ipfs.infura.io`
-- ``IPFSGatewayBackend``
+- ``IPFSGatewayBackend`` (temporarily deprecated)
     - `https://ipfs.io/ipfs/`
 - ``LocalIPFSBacked``
     - connects to a local IPFS API gateway running on port 5001.
 - ``DummyIPFSBackend``
     - Won't pin/fetch files to an actual IPFS node, but mocks out this behavior.
 
-.. py:method:: BaseIPFSBackend.can_resolve_uri(uri)
+.. py:method:: BaseIPFSBackend.pin_assets(file_or_directory_path)
 
-   Returns a bool indicating whether or not this backend is capable of handling the given URI.
+   Pins asset(s) found at the given path and returns the pinned asset data.
 
-.. py:method:: BaseIPFSBackend.fetch_uri_contents(uri)
 
-   Fetches the contents stored at a URI.
+HTTP
+----
 
-.. py:method:: BaseIPFSBackend.pin_assets(file_or_directory_path)
+``Py-EthPM`` offers a backend to fetch files from Github, ``GithubOverHTTPSBackend``.
+
+A valid Github URI *should* conform to the following scheme.
+
+.. code:: python
+
+   https://raw.githubusercontent.com/user/repo/commit_hash/path/to/manifest.json#content_hash
+
+To generate a valid Github PM URI.
+
+- Go to the target manifest in your browser.
+- Press ``y`` to generate the permalink in the address bar.
+- Replace ``"github"`` with ``"raw.githubusercontent"``, and remove the ``"blob"`` namespace from the URI.
+- Suffix the URI with ``#`` followed by the ``keccak`` hash of the bytes found at the Github URI.
```

Comment:
Not for this PR, but it'd be pretty cool to provide a utility on the `GithubOverHTTPSBackend` that did this automatically with a uri and supported a handful of formats.

```python
>>> GithubOverHTTPSBackend.get_permalink_with_content_hash('https://github.com/ethpm/ethpm-spec/blob/master/examples/owned/1.0.0.json')
"https://raw.githubusercontent.com/ethpm/ethpm-spec/3945c47dedb04930ee12c0281494a1b5bdd692a0/examples/owned/1.0.0.json#0x01cbc2a69a9f86e9d9e7b87475e2ba2619404dc8d6ee3cb3a8acf3176c2cace1"
```

Should be able to:

* parse the repository name and file path and branch
* use the github API to pull the latest commit hash
* generate the `raw.github.com` URL
* pull the file and compute the `keccak` of the contents.

This would be something well defined enough that it could be bountied.